File: 2. Medium/0038. Count and Say.py
def countAndSay(self, n: int) -> str:
    if n == 1:
        return "1"
    
    anterior = self.countAndSay(n - 1)
    n = len(prev)
    resultado = ""
    contador = 1
    
    for i in range(n):
        if (i == (n - 1)) or (anterior[i] != anterior[i + 1]):
            resultado += str(contador) + anterior[i]
            contador = 1
        else:
            contador += 1
    
    return resultado

File: 2. Medium/0038. Count and Say.java
class Solution {
    public String countAndSay(int n) {
        String valor = "1";
        for (int i = 0 ; i < n-1 ; i++){
            char caracter = valor.charAt(0);
            StringBuilder resultado = new StringBuilder();
            int contador = 1;
            for (int j = 1 ; j < valor.length() ; j++){
                if (caracter != valor.charAt(j)){
                    resultado.append(contador);
                    resultado.append(caracter);
                    contador = 0;
                    caracter = valor.charAt(j);
                }
                contador++;
            }
            resultado.append(contador);
            resultado.append(caracter);
            valor = resultado.toString();
        }
        return valor;
    }
}
